lib/Analysis/OptimizeRelinearizationAnalysis/OptimizeRelinearizationAnalysis.cpp

```diff
@@ -67,6 +61,32 @@ bool hasCiphertextType(ValueRange range) {
 LogicalResult OptimizeRelinearizationAnalysis::solve() {
   math_opt::Model model("OptimizeRelinearizationAnalysis");
 
+  // Using SelectVariableNames here provides some loose coupling between
+  // what the input to this analysis would be when pretty-printing the IR
+  // and the variable names chosen for each op in the ILP.
+  //
+  // If the pass option use-loc-based-variable-names is set, then the variable
+  // names will use the op's Location attribute. This should only be set when
+  // --optimize-relinearization is the only pass applied, as otherwise the loc
+  // is not guaranteed to be unique and this analysis may fail. This is useful
+  // when debugging, as a failing IR can be printed before running this pass in
+  // isolation.
+  SelectVariableNames variableNames = SelectVariableNames(opToRunOn);
+  int resultFreeVarNameIndex = 0;
+  auto uniqueName = [&](Operation *op) {
+    std::string varName;
+    llvm::raw_string_ostream ss(varName);
+    ss << op->getName() << "_";
+    if (useLocBasedVariableNames) {
+      ss << op->getLoc();
+    } else if (op->getNumResults() == 0) {
+      ss << resultFreeVarNameIndex++;
+    } else {
```

Comment:
QQ - why not only use the free var name index if you have to use it for the 0 result case? Can you just omit SelectVariableNames in this pass completely if you already have a free var name counter?
(I guess for simplicity why not apply this to all ops, regardless of how many results they have).

Reply:
Yeah, I could just give up and use completely opaque identifiers when the pass option is not set. I think I will do that.